addons/storysource/src/loader/default-options.js

```diff
@@ -5,6 +5,7 @@ const defaultOptions = {
     bracketSpacing: true,
     trailingComma: 'es5',
     singleQuote: true,
+    parser: 'babylon',
```

Comment:
We actually [add parser option](https://github.com/storybooks/storybook/blob/master/addons/storysource/src/loader/generate-helpers.js#L33-L38) on call site

So we could just remove `parser !== 'javascript'` check, and replace `'javascript'` string with `'babylon'` in rest of the addon codebase

But I think a more elegant solution would be to pass `filepath` option to prettier so that it could infer the correct parser from extension. You can get file path inside [transform function](https://github.com/storybooks/storybook/blob/master/addons/storysource/src/loader/index.js#L6) as [`this.resourcePath`](https://webpack.js.org/api/loaders/#this-resourcepath), add it to options object, and then pass to prettier instead of parser

Reply:
Both solutions would fix https://github.com/storybooks/storybook/pull/3660#issuecomment-392971907

Reply:
Thank you for information! I will try it.

Reply:
I fixed this.
Pass `this.resourcePath` to prettier config.

Thank you for telling me the information.
It was very helpful in fixing it.